### PR TITLE
SQL: Prevent NPE when getting indices from map container

### PR DESCRIPTION
**Pull request by Fly-Style**
EE counterpart: https://github.com/hazelcast/hazelcast-enterprise/pull/4664

Fixes https://github.com/hazelcast/hazelcast/issues/20600

### CODE REVIEW

**Review by ivanthescientist — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/sql/impl/schema/map/MapTableUtils.java [hunk 1]
```diff
@@ -75,6 +75,10 @@ public static long estimatePartitionedMapRowCount(NodeEngine nodeEngine, MapServ
     public static List<MapTableIndex> getPartitionedMapIndexes(MapContainer mapContainer, List<TableField> fields) {
         Map<QueryPath, Integer> pathToOrdinalMap = mapPathsToOrdinals(fields);
 
+        if (mapContainer.getIndexes() == null) {
```
Comment (promanenko) [inline hazelcast/src/main/java/com/hazelcast/sql/impl/schema/map/MapTableUtils.java:78]: shouldn't we try to get indexes from partitions here?
like in https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java#L338

**Comment by promanenko** [comment]
there is one more place with NPE thrown on "suggest mapping" request:
https://github.com/hazelcast/hazelcast/blob/master/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/MetadataResolver.java#L67

File: hazelcast/src/main/java/com/hazelcast/sql/impl/schema/map/MapTableUtils.java (hunk 1 above)
Comment (Fly-Style) [reply hazelcast/src/main/java/com/hazelcast/sql/impl/schema/map/MapTableUtils.java:78]: @promanenko This method is supposed to find local indices. SQL reads only global indices.

**Review by promanenko — APPROVED** [review]

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/MetadataResolver.java [hunk 2]
```diff
@@ -64,6 +64,11 @@ public Mapping resolve(String iMapName) {
     @Nullable
     @SuppressWarnings("rawtypes")
     private Metadata resolveFromHd(MapContainer container) {
+        if (container.getIndexes() == null) {
+            return null;
+
```
Comment (viliam-durina) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/MetadataResolver.java:69]: Extra line
```suggestion
```

**Review by viliam-durina — APPROVED** [review]

**Pull request merged**